app-react/src/jsMain/kotlin/com/ustadmobile/view/message/conversationlist/ConversationListScreen.kt

```diff
@@ -0,0 +1,170 @@
+
+    var uiState: ConversationListUiState
+
+    var onClickEntry: (MessageAndOtherPerson) -> Unit
+
+}
+
+
+private val ConversationListScreenComponent2 = FC<ConversationListScreenProps> { props ->
+
+    val infiniteQueryResult = usePagingSource(
+        props.uiState.conversations, true, 50
+    )
+    val muiAppState = useMuiAppState()
+
+    VirtualList {
+        style = jso {
+            height = "calc(100vh - ${muiAppState.appBarHeight}px)".unsafeCast<Height>()
+            width = 100.pct
+            contain = Contain.strict
+            overflowY = Overflow.scroll
+        }
+
+        content = virtualListContent {
+
+            infiniteQueryPagingItems(
+                items = infiniteQueryResult,
+                key = { (it.message?.messageUid).toString() }
+            ) { messageItem ->
+                ConversationItem.create {
+                    message = messageItem
+                    onListItemClick = props.onClickEntry
+                }
+            }
+        }
+
+        Container {
+            VirtualListOutlet()
+        }
+    }
+}
+
+val demoConversationList = (0..150).map {
+    MessageAndOtherPerson().apply {
+        message = Message().apply {
+            messageText = "The Conversation number $it"
+        }
+        otherPerson = Person().apply {
+            firstNames = "Person $it"
+            lastName = "$it"
+            personUid = it.toLong()
+        }
+    }
+}
+
+val ConversationListScreenPreview = FC<Props> {
+
+
+    val conversationListUiState by useState {
+        ConversationListUiState(
+            conversations = { ListPagingSource(demoConversationList) }
+        )
+    }
+
+    ConversationListScreenComponent2 {
+        uiState = conversationListUiState
+    }
+}
+
+val ConversationListScreen = FC<Props> {
+    val viewModel = useUstadViewModel { di, savedStateHandle ->
+        ConversationListViewModel(di, savedStateHandle)
+    }
+    val uiStateVal by viewModel.uiState.collectAsState(ConversationListUiState())
+    val appState by viewModel.appUiState.collectAsState(AppUiState())
+
+    UstadFab {
+        fabState = appState.fabState
+    }
+
+    ConversationListScreenComponent2 {
+        uiState = uiStateVal
+        onClickEntry = viewModel::onClickEntry
+    }
+}
+
+external interface ConversationItemProps: Props {
+
+    var message: MessageAndOtherPerson?
+
+    var onListItemClick: (MessageAndOtherPerson) -> Unit
+
+}
+
+val ConversationItem = FC<ConversationItemProps> { props ->
+
+    val formattedTime = useFormattedTime(
+        timeInMillisSinceMidnight = (props.message?.message?.messageTimestamp ?: 0).toInt(),
+    )
+
+    ListItem {
+        ListItemButton{
+            onClick = {
+                props.message?.also { props.onListItemClick(it) }
+            }
+
+            ListItemIcon {
+                UstadPersonAvatar {
+                    personUid = props.message?.otherPerson?.personUid ?: 0
```

Comment:
UstadPersonAvatar actually doesn't need personUid anymore. You need to pass the personName and pictureUri.